shaders/mobile_nerf/mlp_combo.frag

```diff
@@ -0,0 +1,238 @@
+/* Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * ------------------------------------------------------------------------
+ *
+ * THIS IS A MODIFIED VERSION OF THE ORIGINAL FILE
+ *
+ * The original file, along with the original Apache-2.0 LICENSE can be found at:
+ * https://github.com/google-research/jax3d/tree/main/jax3d/projects/mobilenerf
+ *
+ * Modification details: Shader code was updated to work on Vulkan (originally
+ * built for WebGL)
+ * Contributor: (Qualcomm) Rodrigo Holztrattner - quic_rholztra@quicinc.com
+ */
+#version 460
+#extension GL_EXT_nonuniform_qualifier : enable
+
+layout (input_attachment_index = 0, binding = 0) uniform subpassInput inputFeature_0;
+layout (input_attachment_index = 1, binding = 1) uniform subpassInput inputFeature_1;
+layout (input_attachment_index = 2, binding = 2) uniform subpassInput rayDirectionIn;
+layout (input_attachment_index = 3, binding = 3) uniform usubpassInput weightsIndex;
+
+layout(location = 0) out vec4 o_color;
+
+// Try defining constants in the shader itself
+precision highp float;
+
+#define WEIGHTS_0_COUNT (176)
+#define WEIGHTS_1_COUNT (256)
+// The third layer's size is changed from 48 to 64 to make sure a 16 bytes alignement
+//#define WEIGHTS_2_COUNT (48)
+#define WEIGHTS_2_COUNT (64)
+#define BIAS_0_COUNT (16)
+#define BIAS_1_COUNT (16)
+// The third layer bias' size is changed from 3 to 4 to make sure a 16 bytes alignement
+#define BIAS_2_COUNT (4)
+layout(binding = 4) uniform mlp_weights
+{
+	vec4 data[(WEIGHTS_0_COUNT + WEIGHTS_1_COUNT + WEIGHTS_2_COUNT +
+               BIAS_0_COUNT + BIAS_1_COUNT + BIAS_2_COUNT)/4];        // Array of floats
+} weights_arr[];
+
+vec3 evaluateNetwork(  vec4 f0,  vec4 f1,  vec4 viewdir,  uint idx) {
+
+        vec3 res;
+
+        int bias_0_ind = WEIGHTS_0_COUNT + WEIGHTS_1_COUNT + WEIGHTS_2_COUNT;
+        vec4 intermediate_one[4] = vec4[](
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 1],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 2],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 3]
+        );
+
+        #if 0
```

Comment:
Is this #if 0-ed section worth to be kept? Why?

Reply:
Removed these and all other occurences